### DISCUSSION TITLE
RDP Re-negotiation

### DISCUSSION THREAD

**Discussion by AhmadSoboh** [FreeRDP/FreeRDP#9157] [Q&A]
I would like to ask if it is possible to have the client starting RDP negotiation after already done that? I have a pcap for RDP traffic where the client and the server finished SSL handshake and started exchanging encrypted traffic and then suddenly the client sent a RDP negotiation request! Is that valid? I didn't find anything in RDP spec.
Also the server replied to that negotiation request and both started a new SSL handshake...
All of these happened in the same TCP session

**Comment by hardening** [comment]
RDP renegociation is perfectly legit, and it's detailed in the main RDP spec (2.2.3 Deactivation-Reactivation Sequence), that's mostly used when the server wants to resize the client.
I had never noticed that it was matching a SSL renegociation too, or is it a new SSL handshake ? What are running your 2 peers (mstsc -> windows server ?) ?

**Comment by AhmadSoboh** [comment]
@hardening thanks for your reply
It is a new SSL handshake.
It is from Microsoft RDP client for mac -> Window server
Any idea how can I reproduce this behavior from a Windows machine and not Mac?
